vm/vm.go

```diff
@@ -1128,6 +1141,46 @@ func (vm *VM) loadAcceptedBlocks(ctx context.Context) error {
 	return nil
 }
 
+func (vm *VM) restoreAcceptedQueue(ctx context.Context) error {
+	has, err := vm.HasLastProcessed()
+	if err != nil {
+		return errors.New("could not determine if have last processed")
+	}
+	if !has {
+		return nil
+	}
+
+	lastProcessedHeight, err := vm.GetLastProcessedHeight()
+	if err != nil {
+		return fmt.Errorf("could not get last processed height: %w", err)
+	}
+
+	start := lastProcessedHeight + 1
+	end := vm.lastAccepted.Height()
+	if end >= start {
```

Comment:
Why do we terminate here if `end >= start` but we do one more iteration within the for loop when `height == end` ?

Reply:
Wow that's a good point, a bug that I introduced. Replaced the `>=` by a `>`. Thanks!

Reply:
Actually the order was reversed!